apps/src/StudioApp.js

```diff
@@ -645,6 +697,53 @@ StudioApp.prototype.handleSharing_ = function (options) {
   }
 };
 
+StudioApp.prototype.renderShareFooter_ = function(container) {
+  var footerDiv = document.createElement('div');
+  footerDiv.setAttribute('id', 'footerDiv');
+  container.appendChild(footerDiv);
+
+  var reactProps = {
+    i18nDropdown: '',
+    copyrightInBase: false,
+    copyrightStrings: copyrightStrings,
+    baseMoreMenuString: window.dashboard.i18n.t('footer.built_on_code_studio'),
+    baseStyle: {
+      paddingLeft: 0,
+      width: $("#visualization").width(),
+    },
+    className: 'dark',
+    menuItems: [
+      {
+        text: window.dashboard.i18n.t('footer.try_hour_of_code'),
+        link: 'https://code.org/learn',
+        newWindow: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.how_it_works'),
+        link: location.href + "/edit",
+        newWindow: false
+      },
+      {
+        text: window.dashboard.i18n.t('footer.copyright'),
+        link: '#',
+        copyright: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.tos'),
+        link: "https://code.org/tos",
+        newWindow: true
+      },
+      {
+        text: window.dashboard.i18n.t('footer.privacy'),
+        link: "https://code.org/privacy",
+        newWindow: true
+      }
+    ]
+  };
+
+  window.dashboard.footer.render(React, reactProps, footerDiv);
```

Comment:
We call this in a few different places now, with largely the same props in each (but some differences). It feels like there should be a cleaner way of sharing this config instead of repeating it everywhere, though it's not immediately clear to me what this way is.

Part of this weirdness comes from crossing the boundaries between dashboard and apps. Some of it comes from differentiating between JS (which is cached across languages) and .haml (which is not).

This is mostly me just thinking out loud. I'm not sure I have any great suggestions for you :-/